### PR TITLE
vim: helpful additions

### LINKED ISSUE #3779: TLDR for Vim has too many tips about things inside Vim
When I look at `tldr vim`, I see that it has many tips that tell you how to do things in Vim, rather than using the `vim` command itself.  I feel like there are more tips that we can add that are really useful. It's already at its max (of 8) items.

What should we do?

### PR DESCRIPTION
**Pull request by ryanolsonx**
- [X] The page has 8 or fewer examples.
- [X] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [X] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).

Closes #3779

### CODE REVIEW

File: pages/common/vim.md [hunk 1]
```diff
@@ -1,16 +1,28 @@
 # vim
 
-> Vi IMproved, a programmer's text editor, provides several modes for different kinds of text manipulation.
+> Vim (Vi IMproved), a programmer's command line text editor, provides several modes for different kinds of text manipulation.
```
Comment (zdroid) [inline pages/common/vim.md]: `command-line` here

File: pages/common/vim.md [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
+`<Esc>:q!<Enter>`
+
+- Open a file at a specified line number:
+
+`vim +{{line number}} {{path/to/file}}`
+
+- View help for setting (such as {{set number}}):
 
-`<Esc>i`
+`<Esc>:help '{{setting name}}'`
```
Comment (zdroid) [inline pages/common/vim.md]:
```suggestion
`<Esc>:help '{{setting_name}}'`
```

File: pages/common/vim.md [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
+`vim {{path/to/file}}`
 
-- Enter text editing mode (insert mode):
+- Quit without saving:
+
+`<Esc>:q!<Enter>`
+
+- Open a file at a specified line number:
+
+`vim +{{line number}} {{path/to/file}}`
```
Comment (zdroid) [inline pages/common/vim.md]:
```suggestion
`vim +{{line_number}} {{path/to/file}}`
```

**Comment by ryanolsonx** [comment]
Thanks for the requested changes.  I'll fix all of those up.

After adding my tips, Vim now has 10 tips.  What do we do about that?

**Comment by zdroid** [comment]
2 of the examples have to be removed, as 8 is the limit. For example, `Open without your custom configurations` and `View help for setting (such as {{set number}}):` appear less useful than the other examples. However, you should wait for other opinions on removing those.

**Comment by ryanolsonx** [comment]
I would say that finding vim documentation (using `:help`) is probably one of the most important things to learn when looking at the tldr.

I can agree that the custom configurations one isn't as important.

**Comment by zdroid** [comment]
You may be quite right, as I'm not a vim user at all, so I have no reliable knowledge on that. I'm glad we agreed on one example so far, so you have to find one more to remove.

**Comment by ryanolsonx** [comment]
I have my proposed vim.md file all set now.

File: pages/common/vim.md [hunk 4]
```diff
@@ -1,20 +1,28 @@
...
 - Open a file:
 
-`vim {{file}}`
+`vim {{path/to/file}}`
 
-- Enter text editing mode (insert mode):
+- Save a file:
 
-`<Esc>i`
+`:write`
```
Comment (zergov) [inline pages/common/vim.md]: Would it be better to omit the `<Esc>` key like this command, for all commands in this file? 
I feel like they should all include it or not include the `<Esc>` at all.

I feel like putting the `<Esc>`key in those definitions assumes that the user is currently in a mode other than `normal`. I feel like only the actual `:commands` should be there.

What do you think?

Comment (ryanolsonx) [reply pages/common/vim.md]: I agree with you.  I'll make the change in my PR.

File: pages/common/vim.md [hunk 5]
```diff
@@ -1,37 +1,37 @@
...
+`:quit!<Enter>`
 
-- Undo the last operation:
+- Open a file at a specified line number:
 
-`<Esc>u`
+`vim +{{line_number}} {{path/to/file}}`
 
-- Search for a pattern in the file (press `n`/`N` to go to next/previous match):
+- View Vim's help manual:
```
Comment (sbrl) [inline pages/common/vim.md]: Perhaps this should be the second example?

Comment (ryanolsonx) [reply pages/common/vim.md]: Great idea. I'll change that in the morning.

**Comment by ryanolsonx** [comment]
Is this ready for merge?  Can I get another reviewer's approval?

**Comment by ryanolsonx** [comment]
@sbrl Thank you!

**Pull request merged**